### ISSUE TITLE
Cholesky pivoting

### ISSUE THREAD

**Issue by SWotherspoon** [bwlewis/GLM#3]
In the sparse and incremental variants, the resulting linear system is solved by Cholesky decomposition with pivoting.  So $X^T X \beta = X^T y$ is solved essentially with

```r
U <- chol(crossprod(X),pivot=TRUE)
p <- attr(U,"pivot")
backsolve(U,backsolve(U,crossprod(X,y)[p],transpose=TRUE))[p]
```

But shouldn't the final line be
```r
backsolve(U,backsolve(U,crossprod(X,y)[p],transpose=TRUE))[order(p)]
```
to invert the original permutation?

**Comment by bwlewis** [comment]
Yes, indeed, the pivoting has this bug sorry.  I am in th emiddle of substantially revising these notes and will include a fix along these in the revised version, which should appear sometime in July 2018.  Thanks!

(sorry for the extreme latency)
